### ISSUE TITLE
[CRITICAL]Unable to find meetId in the page

### ISSUE THREAD

**Issue by pascal-mueller** [Battleman/zoomdl#48]
**Prerequisite**

- [x] I have updated to the latest version,
- [x] (If applicable) I have ensured the password is correct with the flag `-v 0`,
- [x] I have searched the opened and closed issues for a duplicate,

**Describe the bug**
I run `zoomdl -u myurl -p 'mypw' --cookies cookies.txt` which worked yesterday but then I suddendly started getting 
`[CRITICAL]Unable to find meetId in the page`

**Versions (please complete the following information):**
 - OS: Arch
 - Python Version: 3.9.2

**Comment by Battleman** [comment]
Oh, interesting. So from my tests, putting the `-p` AND the `--cookies` option will trigger an error.

Quick explanation, but I should fix that, it's bad behaviour. So when you put cookies, it immediately updates the cookies of the session. And when you give the password, it supposes the video is password-protected and looks for the `meetId` on the page. But this `meetId` is only available on the login page. And because you have the login cookies, it never goes to the login page. 

tl;dr: I'll adapt the doc and behaviour to something more normal. Meanwhile, if you use `-p` OR `--cookies` but not both, it should work. `--cookies` is more 'general' and should bypass all securities, so if in doubt, use that one.

**Comment by pemmenegger** [comment]
Same here but neither using only ```-p``` nor only ```--cookies``` worked for me. Only ```-p``` returned ```[CRITICAL]Unable to find meetId in the page``` and only ```--cookies``` resulted in ```ERROR] Video not found in page. Is it login-protected?```

**Versions (please complete the following information):**
OS: macOS 11.2.3
Python Version: 3.9.0

**Comment by Battleman** [comment]
OK, then this I can't reproduce. Does this happen on all videos? Are there special things about those videos? SSO, password,...?

**Comment by pemmenegger** [comment]
The videos I've tried are all SSO and password protected. Does this help?

**Comment by Battleman** [comment]
I unfortunately can't test that, as I don't have an authorized account to view that. I'll see what I can do, but with no testing it's not gonna be easy, I'll probably need you to guide me.

**Comment by Oorha** [comment]
Battleman, appreciate all you're doing with this.

Unfortunately have the same issue and the link requires an authorised account.

[CRITICAL]Unable to find meetId in the page

How could I help resolve this?

**Comment by Battleman** [comment]
For now, the best method is to use cookies. Does it not work for you?

**Comment by Oorha** [comment]
Cookies works for me, thank you!

**Comment by hykilpikonna** [comment]
I think I found the problem. For a recording with both account and password protection, entering the password in your browser will unlock the video for the account, so `zoomdl` can download it successfully with only cookie and without a password. However, if the video haven't been previously unlocked, using `zoomdl` would fail either with only cookie, or with both cookie and password.
